Worksheet B - Pseudocode.txt

```diff
@@ -0,0 +1,17 @@
+Import random library
+N = Random INT from 1-10
+
+Word = Random String of N Characters
+
+Guess = input ("What is the secret word, if it has " + N + "letters?")
+
+For N until Guess[N] or until after fourth guess:
+	If Guess == Word:
+		Print "You have done it correctly!"
+	If Guess == Word - 1 character:
+		Print "You had a likeness of N-1 characters!"
+	If Guess == Word - 2 characters:
+		Print "You had a likeness of N-2 characters!"
+	...
+	Else:
+		Print "You have failed to correctly guess the word..."
```

Comment:
Flowchart looks mostly OK. The user guessing the word should be included in the loop, otherwise the user will not be given the opportunity to guess again after getting it wrong once (the system will repeatedly check the same guess). Tracing through your flowchart manually would help catch issues like this, and is also a useful skill to develop for debugging code.